formatter/src/main/java/org/togetherjava/tjbot/formatter/formatting/TokenQueue.java

```diff
@@ -0,0 +1,139 @@
+package org.togetherjava.tjbot.formatter.formatting;
+
+import org.togetherjava.tjbot.formatter.tokenizer.Token;
+import org.togetherjava.tjbot.formatter.tokenizer.TokenType;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Queue that holds tokens to be consumed.
```

Comment:
The term "token" is very often used, but I haven't noticed much of a description.

I'd heavily appreciate it if you link to the Token class

Reply:
thats fair. i guess it was clear due to the context of lexxing (= tokenization).

Reply:
done

added some extra paragraph and links